### PR TITLE
add distribute function

### PR DESCRIPTION
**Pull request by pingyourid**
a category help masonry to complete distribute on one direction.

and i have a demo in the url:https://github.com/pingyourid/MasonryHelper

### CODE REVIEW

**Comment by robertjpayne** [comment]
@pingyourid hey it looks pretty good, not sure we're too keen on adding any more API into Masonry as it's now in bugfix only.

Regardless you have a few typos in code such as `AxisTypeHorizon` vs `AxisTypeHorizontal` we'd also want those prefixed with `MAS` to avoid ObjC namespacing issues.

Secondly, we'd need test coverage on these to merge it.

Cheers!

**Comment by pingyourid** [comment]
gladly hear from you,I will do it latter,and also make some example!

**Comment by pingyourid** [comment]
I have finished writing my code,can any owner review it if convenience?

File: Masonry/NSArray+MASAdditions.h [hunk 1]
```diff
@@ -10,6 +10,11 @@
 #import "MASConstraintMaker.h"
 #import "MASViewAttribute.h"
 
+typedef NS_ENUM(NSUInteger, MASAxisType) {
+    MASAxisTypeHorizon,
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: This should read `MASAxisTypeHorizontal` to be consistent with the vertical entry.

File: Masonry/NSArray+MASAdditions.h [hunk 2]
```diff
@@ -42,6 +47,26 @@
  *
  *  @return Array of created/updated MASConstraints
  */
-- (NSArray *)mas_remakeConstraints:(void(^)(MASConstraintMaker *make))block;
+- (NSArray *)mas_remakeConstraints:(void (^)(MASConstraintMaker *make))block;
+
+/**
+ *  distribute with fixed spaceing
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Fix typo `spaceing` should be `spacing`

File: Masonry/NSArray+MASAdditions.h [hunk 3]
```diff
@@ -42,6 +47,26 @@
...
+ *
+ *  @param axisType     horizon/vertical
+ *  @param paddingSpace space
+ *  @param leadSpacing  head
+ *  @param tailSpacing  tail
+ */
+- (void)mas_distributeViewsAlongAxis:(MASAxisType)axisType withFixedSpacing:(CGFloat)paddingSpace leadSpacing:(CGFloat)leadSpacing tailSpacing:(CGFloat)tailSpacing;
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Rename `paddingSpace` to `fixedSpacing` and update the apple doc entry to reflect

File: Masonry/NSArray+MASAdditions.h [hunk 4]
```diff
@@ -42,6 +47,26 @@
...
+
+/**
+ *  distribute with fixed item size
+ *
+ *  @param axisType    horizon/vertical
+ *  @param itemLength  item size
+ *  @param leadSpacing head
+ *  @param tailSpacing  tail
+ */
+- (void)mas_distributeViewsAlongAxis:(MASAxisType)axisType withFixedItemLength:(CGFloat)itemLength leadSpacing:(CGFloat)leadSpacing tailSpacing:(CGFloat)tailSpacing;
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Rename `itemLength` to `fixedItemLength` and update the apple doc entry to reflect

File: Masonry/NSArray+MASAdditions.h [hunk 5]
```diff
@@ -42,6 +47,26 @@
...
  *  @return Array of created/updated MASConstraints
  */
-- (NSArray *)mas_remakeConstraints:(void(^)(MASConstraintMaker *make))block;
+- (NSArray *)mas_remakeConstraints:(void (^)(MASConstraintMaker *make))block;
+
+/**
+ *  distribute with fixed spaceing
+ *
+ *  @param axisType     horizon/vertical
+ *  @param paddingSpace space
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Update apple doc to read `the spacing between each item`

File: Masonry/NSArray+MASAdditions.h [hunk 6]
```diff
@@ -42,6 +47,26 @@
...
+ *  @param leadSpacing  head
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Update apple doc to read `the spacing before the first item and the container`

File: Masonry/NSArray+MASAdditions.h [hunk 7]
```diff
@@ -42,6 +47,26 @@
...
+ *  @param tailSpacing  tail
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Update the apple doc to read the spacing after the last item and the container

File: Masonry/NSArray+MASAdditions.h [hunk 8]
```diff
@@ -42,6 +47,26 @@
  *
  *  @return Array of created/updated MASConstraints
  */
-- (NSArray *)mas_remakeConstraints:(void(^)(MASConstraintMaker *make))block;
+- (NSArray *)mas_remakeConstraints:(void (^)(MASConstraintMaker *make))block;
+
+/**
+ *  distribute with fixed spaceing
+ *
+ *  @param axisType     horizon/vertical
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Update apple doc to read `which axis to distribute items along`

File: Masonry/NSArray+MASAdditions.h [hunk 9]
```diff
@@ -42,6 +47,26 @@
...
+ *  @param paddingSpace space
+ *  @param leadSpacing  head
+ *  @param tailSpacing  tail
+ */
+- (void)mas_distributeViewsAlongAxis:(MASAxisType)axisType withFixedSpacing:(CGFloat)paddingSpace leadSpacing:(CGFloat)leadSpacing tailSpacing:(CGFloat)tailSpacing;
+
+/**
+ *  distribute with fixed item size
+ *
+ *  @param axisType    horizon/vertical
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Update apple doc to read `which axis to distribute items along`

File: Masonry/NSArray+MASAdditions.h [hunk 10]
```diff
@@ -42,6 +47,26 @@
...
+ *  @param itemLength  item size
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Update apple doc to read `the fixed length of each item`

File: Masonry/NSArray+MASAdditions.h [hunk 11]
```diff
@@ -42,6 +47,26 @@
...
+ *  @param leadSpacing head
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Update apple doc to read `the spacing before the first item and the container`

File: Masonry/NSArray+MASAdditions.h [hunk 12]
```diff
@@ -42,6 +47,26 @@
...
+ *  @param tailSpacing  tail
```
Comment (robertjpayne) [inline Masonry/NSArray+MASAdditions.h]: Update apple doc to read `the spacing after the last item and the container`

**Comment by robertjpayne** [comment]
@pingyourid I've added a bunch of comments. Mostly are just some typos and language stuff.

**Comment by pingyourid** [comment]
thank you for teach me English hand by hand..

**Comment by robertjpayne** [comment]
@pingyourid thanks again for tidying this up, merging it -- will probably do a release in a week or so once iOS 9 releases

**Pull request merged**